sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/linearRegression.scala

```diff
@@ -271,8 +271,13 @@ case class RegrSlope(left: Expression, right: Expression) extends DeclarativeAgg
 
   override lazy val initialValues: Seq[Expression] = covarPop.initialValues ++ varPop.initialValues
 
-  override lazy val updateExpressions: Seq[Expression] =
-    covarPop.updateExpressions ++ varPop.updateExpressions
+  override lazy val updateExpressions: Seq[Expression] = {
+    val isNull = left.isNull || right.isNull
+    val updateResult = covarPop.updateExpressions ++ varPop.updateExpressions
+    aggBufferAttributes.zip(updateResult).map { case (oldValue, newValue) =>
+      If(isNull, oldValue, newValue)
```

Comment:
we only need to add this for `varPop.updateExpressions`

Reply:
the code should be something like
```
covarPop.updateExpressions ++ varPop.updateExpressions.zip(varPop.aggBufferAttributes)...
```

Reply:
BTW to add a bit more color to my proposal: the update expressions of `VariancePop` is
```
    trimHigherOrder(Seq(
      If(child.isNull, n, newN),
      If(child.isNull, avg, newAvg),
      If(child.isNull, m2, newM2),
      If(child.isNull, m3, newM3),
      If(child.isNull, m4, newM4)
    ))
```

The `child` appears not only in the if conditions, but also `newAvg`, `newM2`, etc. It's better to not add the extra full check to the `child`, as it will be duplicated.

Reply:
Good insight.